core/src/test/resources/query-tests.yaml

```diff
@@ -460,3 +460,36 @@ tests:
         "port": 8443,
         "scheme": "https"
       }
+
+# parse-url tests with throwOnFailure set to true should behave as above
+
+  - input:
+    query: parse-url("https://example.com:8443", true)
+    output: |
+      {
+        "host": "example.com",
+        "port": 8443,
+        "scheme": "https"
+      }
+
+# parse-url tests with throwOnFailure set to false should behave as above for good cases
+
+  - input:
+    query: parse-url("https://example.com:8443", false)
+    output: |
+      {
+        "host": "example.com",
+        "port": 8443,
+        "scheme": "https"
+      }
+
+# parse-url tests with throwOnFailure set to false should return an error object for bad cases
+
+  - input:
+    query: parse-url("this-is-an-invalid-url", false)
+    output: |
+      {
+        "error": "MalformedURLException",
+        "message": "no protocol: this-is-an-invalid-url",
+        "input": "this-is-an-invalid-url"
+      }
```

Comment:
We need a test of what happens if the second argument is not a boolean. IMHO that should be considered an error: ie, throw an exception.

We also need a test of `null` handling, with and without the second parameter. (Should have had a `null` test already, sorry.) I think `null` input should lead to `null` output, regardless of the value of the second parameter.

Reply:
I am unsure how you typically tests for failures. Do you have an example? I looked in the query-tests.yaml and found no obvious cases to me

Reply:
See `function-error-tests.json` for examples.

Reply:
Added tests